### DISCUSSION TITLE
how to build on windows with openssl and nghttp3?

### DISCUSSION THREAD

**Discussion by peguine** [curl/curl#13701] [Q&A] [answered]
in winbuild\Makefile.vc, there is no option like  '--with-openssl-quic'  and '--with-nghttp3'

**Comment by bagder** [comment]
We recommend using cmake on Windows to build HTTP/3 enabled. As documented in `docs/HTTP3.md`.

**Reply by peguine** [reply]
I have already read that document, but it only covers how to build  on Linux and does not explain how to build on Windows.

**Reply by bagder** [reply]
cmake works fine on Windows as well.

**Comment by peguine** [comment]
solved , manually add the root path of openssl , zlib, nghttp3 in cmakelists.txt
`set(NGHTTP3_ROOT "D:/doc/c++/libs/nghttp3")`

**Comment by jshanab** [comment]
Can you show me your invokation line for CMAke? I am just trying to get curl with openssl for https and do not need http3 but the resulting build says during config https protocol but does not support https in the build.

**Reply by jzakrzewski** [reply, marked as answer]
I literally use this:
```
set PREFIX_DIR=D:/local/%PLATFORM%-%FLAVOUR%
 
cmake -G "Ninja" -DCMAKE_BUILD_TYPE=%FLAVOUR% -DCMAKE_PREFIX_PATH:PATH=%PREFIX_DIR% -DCMAKE_INSTALL_PREFIX:PATH=%PREFIX_DIR% -DBUILD_TESTING:BOOL=OFF -DUSE_WINDOWS_SSPI=ON  -DCURL_USE_OPENSSL=ON -DUSE_NGHTTP2=ON -DCURL_DISABLE_SOCKETPAIR=ON ../curl-%VERSION%
cmake --build . --target install
```
run from the appropriate Visual Studio developer prompt.
All my dependencies are nicely installed under %PREFIX_DIR%, but some . You may want to get rid of some options ;)

**Reply by jshanab** [reply]
Thanks!.  
What version of openssl were you working against.?
I had to modify but the main diff in your invocation is the addition of USE_WINDOWS_SSPI (gotta look that one up). But when built, still the https protocol is not there as the ssl backend is not even mentioned. Even thoug it was required to run the cmake command and the cmake command lists the https protocol as selected. It is acting like the #defines in the c++ code are not being set.

F:\CurlFromSource\local\x64\bin>curl -V
curl 8.9.0-DEV (Windows) libcurl/8.9.0-DEV
Release-Date: [unreleased]
Protocols: dict file ftp gopher http imap ipfs ipns ldap ldaps mqtt pop3 rtsp smb smtp telnet tftp
Features: alt-svc AsynchDNS HSTS IPv6 Largefile NTLM threadsafe UnixSockets

Maybe my openssl 3.1.0a is not compatible with this cmake or curl version.
